server/channels/store/storetest/user_store.go

```diff
@@ -6182,3 +6186,302 @@ func testUpdateLastLogin(t *testing.T, rctx request.CTX, ss store.Store) {
+			},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, userReport)
+		require.Equal(t, 3, len(userReport))
+
+		require.NotNil(t, userReport[0])
+		require.Equal(t, u1.Username, userReport[0].Username)
+
+		require.NotNil(t, userReport[1])
+		require.Equal(t, u2.Username, userReport[1].Username)
+
+		require.NotNil(t, userReport[2])
+		require.Equal(t, u3.Username, userReport[2].Username)
+	})
+
+	t.Run("should return in the correct order", func(t *testing.T) {
+		userReport, err := ss.User().GetUserReport(&model.UserReportOptions{
+			UserReportOptionsWithoutDateRange: model.UserReportOptionsWithoutDateRange{
+				SortColumn: "Username",
+				SortDesc:   true,
+				PageSize:   50,
+			},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, userReport)
+		require.Equal(t, 3, len(userReport))
+
+		require.NotNil(t, userReport[0])
+		require.Equal(t, u3.Username, userReport[0].Username)
+
+		require.NotNil(t, userReport[1])
+		require.Equal(t, u2.Username, userReport[1].Username)
+
+		require.NotNil(t, userReport[2])
+		require.Equal(t, u1.Username, userReport[2].Username)
+	})
+
+	t.Run("should fail on invalid sort column", func(t *testing.T) {
+		userReport, err := ss.User().GetUserReport(&model.UserReportOptions{
+			UserReportOptionsWithoutDateRange: model.UserReportOptionsWithoutDateRange{
+				SortColumn: "FakeColumn",
+				SortDesc:   true,
+				PageSize:   50,
+			},
+		})
+		require.Error(t, err)
+		require.Nil(t, userReport)
+	})
+
+	t.Run("should only return amount of users in page", func(t *testing.T) {
+		userReport, err := ss.User().GetUserReport(&model.UserReportOptions{
+			UserReportOptionsWithoutDateRange: model.UserReportOptionsWithoutDateRange{
+				SortColumn: "Username",
+				PageSize:   2,
+			},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, userReport)
+		require.Equal(t, 2, len(userReport))
+
+		require.NotNil(t, userReport[0])
+		require.Equal(t, u1.Username, userReport[0].Username)
+
+		require.NotNil(t, userReport[1])
+		require.Equal(t, u2.Username, userReport[1].Username)
+	})
+
+	t.Run("should return correct paging", func(t *testing.T) {
+		userReport, err := ss.User().GetUserReport(&model.UserReportOptions{
+			UserReportOptionsWithoutDateRange: model.UserReportOptionsWithoutDateRange{
+				SortColumn:          "Username",
+				PageSize:            50,
+				LastSortColumnValue: u2.Username,
+				LastUserId:          u2.Id,
+			},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, userReport)
+		require.Equal(t, 1, len(userReport))
+
+		require.NotNil(t, userReport[0])
+		require.Equal(t, u3.Username, userReport[0].Username)
+	})
+
+	t.Run("should return accurate post stats for various date ranges", func(t *testing.T) {
+		userReport, err := ss.User().GetUserReport(&model.UserReportOptions{
+			UserReportOptionsWithoutDateRange: model.UserReportOptionsWithoutDateRange{
+				SortColumn: "Username",
+				PageSize:   50,
+			},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, userReport)
```

Comment:
Maybe it's better to add a `len` assertion. (There are similar occurrences below.)